### PR TITLE
Upgrade everything except travis to VS 2019

### PR DESCRIPTION
**Pull request by Margen67**
AppVeyor added support for VS 2019, and it's finally June 1st.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -16,7 +16,7 @@ before_script:
     export PATH="c:\Program Files (x86)\Microsoft Visual Studio\2017\BuildTools\MSBuild\15.0\Bin":$PATH;
```
Comment (Margen67) [inline .travis.yml]: Should this be changed to 2019?

Comment (RadWolfie) [reply .travis.yml]: I originally meant to delete it. I forgot to do so in the previous pull request I made.

Comment (RadWolfie) [reply .travis.yml]: Actually, it need change to 2019.

**Comment by RadWolfie** [comment]
While the doc and CI updates look good. I'm not sure if Travis CI has VS 2019 support or not.

**Comment by x1nixmzeng** [comment]
The distorm lib needs rebuilding for release to compile

**Comment by Margen67** [comment]
I removed the travis commit since travis doesn't support VS2019, and removed the generator from CMakeSettings.json.

**Comment by RadWolfie** [comment]
@x1nixmzeng distorm isn't the main issue. AppVeyor's error is `d3dx9.lib(d3dxvec4.obj) : fatal error LNK1103: debugging information corrupt; recompile module [C:\projects\cxbx-reloaded\build\projects\cxbx\cxbx.vcxproj]` for release build.

Since @Margen67 removed the generator config, the issue has been flip from release to debug build.

Base on the solutions had been attempted, I can only see two options for CMakeSettings.json.

A) Use "Visual Studio 16 2019" generator enforce. (Using Ninja generator didn't work for release build in AppVeyor.)
   * Pro: Known successful compiled for both release and debug builds: https://ci.appveyor.com/project/SoullessSentinel/cxbx-reloaded/builds/24978382
   * Con: Testers/Devs using Visual Studio 2017 can't use File > Open > CMake option directly.

B) Remove CMakeSettings.json file completely.
   * Use terminal to do the steps from the README.md file.
   * Con: Visual Studio will decide where to put build folder at unless have setup a CMakeSettings.json to enforce for every projects allocated with Visual Studio.

---

As for the record had been attempted for CMakeSettings.json' generator:
* "Visual Studio 16 2019" generator (release: ✔️ | debug: ✔️)
   * https://ci.appveyor.com/project/SoullessSentinel/cxbx-reloaded/builds/24978382
* "Ninja" generator (release: ❌ | debug: ✔️)
   * https://ci.appveyor.com/project/SoullessSentinel/cxbx-reloaded/builds/24979057
* No generator given (release: ✔️ | debug: ❌)
   * https://ci.appveyor.com/project/SoullessSentinel/cxbx-reloaded/builds/25188024

**Comment by RadWolfie** [comment]
Since the code does not affect Visual Studio 2017 users, it is safe to merge this.

**Comment by RadWolfie** [comment]
Hmm... After another re-test. It seems C# is not detected with the default generator, Ninja. Way to go CMake... 🙁

**Comment by RadWolfie** [comment]
Found the fault on my end. It was AV software sending the build program to a secure sandbox.

I can confirm removing the generator key does work as intend since Visual Studio is allow to select their generator version (2017 for 2017 and 2019 for 2019). Compiling also works too.

**Comment by RadWolfie** [comment]
Since this pull request allow backward compatible with VS 2017, I'm merging it now.

**Pull request merged**